src/html5/parser.rs

```diff
@@ -2966,6 +2986,22 @@ impl<'chars> Html5Parser<'chars> {
             Token::StartTag { name, .. } if name == "template" => {
                 let node_id = self.insert_html_element(&self.current_token.clone());
 
+                self.active_formatting_elements_push_marker();
```

Comment:
More work needs to be done. Html5ever doesn't seem to do this, but I think we should